dotnet/src/Connectors/Connectors.OpenAIV2/OpenAIAudioToTextExecutionSettings.cs

```diff
@@ -0,0 +1,197 @@
+    /// Default is 0.
+    /// </summary>
+    [JsonPropertyName("temperature")]
+    public float Temperature
+    {
+        get => this._temperature;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._temperature = value;
+        }
+    }
+
+    /// <summary>
+    /// The timestamp granularities to populate for this transcription. response_format must be set verbose_json to use timestamp granularities. Either or both of these options are supported: word, or segment.
+    /// </summary>
+    [JsonPropertyName("granularities")]
+#pragma warning disable CA1819 // Properties should not return arrays
+    public TimeStampGranularities[]? Granularities { get; set; }
+#pragma warning restore CA1819 // Properties should not return arrays
+
+    /// <summary>
+    /// Creates an instance of <see cref="OpenAIAudioToTextExecutionSettings"/> class with default filename - "file.mp3".
+    /// </summary>
+    public OpenAIAudioToTextExecutionSettings()
+        : this(DefaultFilename)
+    {
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="OpenAIAudioToTextExecutionSettings"/> class.
+    /// </summary>
+    /// <param name="filename">Filename or identifier associated with audio data. Should be in format {filename}.{extension}</param>
+    public OpenAIAudioToTextExecutionSettings(string filename)
+    {
+        this._filename = filename;
+    }
+
+    /// <inheritdoc/>
+    public override PromptExecutionSettings Clone()
+    {
+        return new OpenAIAudioToTextExecutionSettings(this.Filename)
+        {
+            ModelId = this.ModelId,
+            ExtensionData = this.ExtensionData is not null ? new Dictionary<string, object>(this.ExtensionData) : null,
+            Temperature = this.Temperature,
+            ResponseFormat = this.ResponseFormat,
+            Language = this.Language,
+            Prompt = this.Prompt
+        };
+    }
+
+    /// <summary>
+    /// Converts <see cref="PromptExecutionSettings"/> to derived <see cref="OpenAIAudioToTextExecutionSettings"/> type.
+    /// </summary>
+    /// <param name="executionSettings">Instance of <see cref="PromptExecutionSettings"/>.</param>
+    /// <returns>Instance of <see cref="OpenAIAudioToTextExecutionSettings"/>.</returns>
+    public static OpenAIAudioToTextExecutionSettings? FromExecutionSettings(PromptExecutionSettings? executionSettings)
+    {
+        if (executionSettings is null)
+        {
+            return new OpenAIAudioToTextExecutionSettings();
+        }
+
+        if (executionSettings is OpenAIAudioToTextExecutionSettings settings)
+        {
+            return settings;
+        }
+
+        var json = JsonSerializer.Serialize(executionSettings);
+
+        var openAIExecutionSettings = JsonSerializer.Deserialize<OpenAIAudioToTextExecutionSettings>(json, JsonOptionsCache.ReadPermissive);
+
+        if (openAIExecutionSettings is not null)
```

Comment:
Similar like in ClientCore, consider suppressing null and remove check. In other FromExecutionSettings method too.